packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { concatStyleSets, IStyle, ITextFieldStyles, mergeStyles, TextField, useTheme } from '@fluentui/react';
+import React, { useState } from 'react';
+import { useLocale } from '../../localization';
+import { containerStyles, digitStyles, rowStyles, subStyles, textFieldStyles } from './Dialpad.styles';
+
+/**
+ * Strings of {@link Dialpad} that can be overridden.
+ *
+ * @public
+ */
+export interface DialpadStrings {
+  errorText: string;
+  defaultText: string;
+}
+
+/**
+ * Props for {@link Dialpad} component.
+ *
+ * @public
+ */
+export interface DialpadProps {
+  strings?: DialpadStrings;
+  containerStyles?: IStyle;
+  rowStyles?: IStyle;
+  textFieldStyles?: Partial<ITextFieldStyles>;
+  digitStyles?: IStyle;
+  subStyles?: IStyle;
+}
+
+/**
+ * A component to allow users to enter phone number through clicking on dialpad/using keyboard
+ *
+ *
+ * @public
+ */
+export const Dialpad = (props: DialpadProps): JSX.Element => {
+  const localeStrings = useLocale().strings.dialpad;
+  const strings = { ...localeStrings, ...props.strings };
+  return <DialpadContainer errorText={strings.errorText} defaultText={strings.defaultText} {...props} />;
+};
+
+const formatPhoneNumber = (value: string, defaultText: string): string => {
```

Comment:
regex and custom formatting is hard to review and fragile, can you add some unit tests for expected formats for this

Reply:
We have great support for north american numbers! But alternatively is there an npm package we can make use of instead of having to do this manually ourselves? Reason is to avoid maintenance and to have a wider support (an open source library is more likely to support more intl number formats than we will think of)

Reply:
After discussions during feature meeting ,we are only supporting NA formatting for now so we will stick with this formatting function. Working on the unit tests now :)

Reply:
updated with unit test